src/configuration.cpp

```diff
@@ -172,6 +179,48 @@ chimera::CompiledConfiguration::CompiledConfiguration(
   , bindingNode_(configNode_["template"]) // TODO: is this always ok?
   , ci_(ci)
 {
+    // This placeholder will be filled in by the options.strict specified
+    // in the configuration YAML if it exists, or remain unset otherwise.
+    boost::optional<bool> config_options_strict;
+
+    // Parse 'options' section of configuration YAML if it exists.
+    if (configNode_)
```

Comment:
Do we have to do this explicitly?  I thought `YAML::Node` would simply return a node of type `Undefined` on a lookup failure, which would allow you to do optional chaining e.g.:

```
const auto &strictNode = configNode_["options"]["strict"];
if (strictNode_ )
{
    [...]
}
```

Reply:
`YAML::Node` throw an exception for lookup from a "zombie" (invalid) node. In this case, `configNode_["options"]` could be a zombie node when "options" tag is missing.

Reply:
Ugh.  Can we make a lookup helper that just something like:
```
YAML::Node& safeLookup(const YAML::Node &node, const char *key)
{
    return node[key];
}

template<typename... Args>
YAML::Node& safeLookup(const YAML::Node &node, const char *key, Args... args)
{
    const auto &next = node[key];
    if (!next) return next;
    safeLookup(next, args...);
}

const auto &result = safeLookup(configNode_. "options", "strict");
```

We don't have to do this if it's too difficult, we can approve as is too.

Reply:
This would go into e.g. https://github.com/personalrobotics/chimera/blob/887dbbc7f30491933f3b004eca136db15677c738/include/chimera/util.h if we implemented it

Reply:
Done in https://github.com/personalrobotics/chimera/pull/225/commits/b40af517af75a8c537d889310dd49bd1af21da0e.

Modified to return invalid node for both function overloads when `node` itself is invalid. Otherwise, those functions would throw an exception.